### PR TITLE
Expose helper functions for validating email addresses and getting multiple email addresses from a string

### PR DESCRIPTION
**Pull request by lfdebrux**
A lot of our scripts send out emails to users based on addresses that they supply. These addresses have historically been collected from a form and stored in the database without any validation of the input. This means that we now have some failures where the user has tried to supply multiple addresses in one string (see [this ticket](https://trello.com/c/DHDgBuns)).

This PR tries to make it so that in future we can change our scripts so that a) they can handle strings that contain multiple email addresses; and b) they can try to validate the email addresses before they call Notify.

It does this by exposing the email validation algorithm that was recently added by @risicle as a helper function, and also by adding a new helper function that splits a string based on a set of characters to generate a list of possible email addresses.

### CODE REVIEW

File: dmutils/email/helpers.py [hunk 1]
```diff
@@ -2,9 +2,105 @@
...
+    ['bob@invalid', 'bob.blob@job.com']
+    >>> email_addresses("bob@blob;bob.blob@job.com;bob@blob..invalid")
+    ['bob@blob', 'bob.blob@job.com', 'bob@blob..invalid']
+    """
+
+    addresses = [string]
+    for sep in separators:
+        if sep in string:
+            addresses = string.split(sep)
+            break  # earlier separators take precedence
```
Comment (Wynndow) [inline dmutils/email/helpers.py]: This is probably pretty unlikely, but what if the string uses different separators? `'bob@blob', 'bob.blob@job.com'; 'bob.bob@blob.com'` for example.
This is probably too much of an edge case to worry about but thought it worth thinking about.

File: tests/forms/test_validators.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
+
+import pytest
+import mock
+
+from wtforms.validators import ValidationError
+
+import dmutils.forms.validators
+
+
+@pytest.fixture
```
Comment (Wynndow) [inline tests/forms/test_validators.py]: Should these live in `conftest.py`? I'm not sure. We seem to create some fixtures in test files and some in `conftest.py`

File: dmutils/email/helpers.py (hunk 1 above)
Comment (lfdebrux) [reply dmutils/email/helpers.py]: I thought about it but decided it was probably too much of an edge case to be worth the additional complexity.

To be fair it should be a fairly straightforward change, basically it would be

```
    for sep in separators:
        if sep in string:
            string = string.split(sep)
```

but I worried it might have weird and hard to debug side-effects.

I guess if something like this appears in the wild we can deal with it then.

File: tests/forms/test_validators.py (hunk 2 above)
Comment (risicle) [reply tests/forms/test_validators.py]: I think we put them in `conftest.py` when they would appear to be more generally useful, but keep them in-test-module when they're rather specific.

Comment (lfdebrux) [reply tests/forms/test_validators.py]: My understanding from the pytest documentation is that fixtures that are shared across modules go in `conftest.py`.

I guess there would be a benefit to having a global email address fixture, currently this specific fixture is used only for my tests, so I don't think it needs to be in `conftest.py` until that changes.
